### PR TITLE
Added Makefile flexibility to use in CI and dev builds

### PR DESCRIPTION
**Pull request by msterin**
1. Made user (for dockerhub) and tag configurable to be used in CI 
2. changed env var to set debug in the plugin from DVV_... to VDVS_...
3. improved makefile info reporting
4. made default plugin tag to be constructed from git current tag

Contributes to #886 
tested: manual runs and demo to @shuklanirdesh82  :-) 

NOTE: Managed plugin are out of "experimental" in  Docker 1.13, so it works on 1.13 or new versioning scheme (Docker '17.xx...') 

//CC @shuklanirdesh82

### CODE REVIEW

File: plugin/Makefile [hunk 1]
```diff
@@ -17,22 +17,36 @@
...
+GIT_SHA := $(shell git rev-parse --revs-only --short HEAD)
+GIT_TAG := $(shell git describe --tags --abbrev=0 $(GIT_SHA))
 
-# Binaries we want to pick up
-BINARY := docker-volume-vsphere
+# Allow PLUGIN_TAG to be supplied via env, or use the latest git tag.
+# use EXTRA_TAG for adding suffixes like "-CI" or "-RELEASE" (otional)
+PLUGIN_TAG ?= $(GIT_TAG)$(EXTRA_TAG)
+
+# plugin name - used as a base for ful plugin name and container for extracting rootfs
```
Comment (shuklanirdesh82) [inline plugin/Makefile]: nit: "full" ... I guess

File: plugin/Makefile [hunk 2]
```diff
@@ -41,17 +55,28 @@ clean:
...
 	@echo "-- Creating  plugin $(PLUGIN_NAME):$(PLUGIN_TAG) ..."
 	docker plugin create $(PLUGIN_NAME):$(PLUGIN_TAG) .
+	@touch $@
 
 
 push:
-	docker plugin push $(PLUGIN_NAME):$(PLUGIN_TAG)
+	@echo Pushing $(PLUGIN_NAME):$(PLUGIN_TAG)  to dockerhub.io...
+	@docker plugin push $(PLUGIN_NAME):$(PLUGIN_TAG) || \
+			echo 'Please make sure the plugin is build ("make clean plugin") and you have logged in to docker hub first ("docker login -u $(DOCKER_HUB_USER)")'
```
Comment (shuklanirdesh82) [inline plugin/Makefile]: how about following message?

```the plugin is build``` => ```the plugin is built```

Comment (msterin) [inline plugin/Makefile]: thanks, both typos fixed

**Comment by msterin** [comment]
CI failing on the known issue `VolumeDriver.Create: Internal issue: ret != 0 but errno is not set. Cancelling operation.`

Merging

File: plugin/Makefile [hunk 3]
```diff
@@ -41,17 +55,28 @@ clean:
...
 	docker rm -vf  $(TMP_CONTAINER)
+	docker rmi $(TMP_IMAGE)
 	@echo "-- Creating  plugin $(PLUGIN_NAME):$(PLUGIN_TAG) ..."
 	docker plugin create $(PLUGIN_NAME):$(PLUGIN_TAG) .
+	@touch $@
 
 
 push:
-	docker plugin push $(PLUGIN_NAME):$(PLUGIN_TAG)
+	@echo Pushing $(PLUGIN_NAME):$(PLUGIN_TAG)  to dockerhub.io...
```
Comment (shaominchen) [inline plugin/Makefile]: Nit: remove one redundant space

**Pull request merged**